extensions/http-curl/processors/InvokeHTTP.cpp

```diff
@@ -205,7 +205,7 @@ void InvokeHTTP::onSchedule(const std::shared_ptr<core::ProcessContext> &context
     logger_->log_debug("%s attribute is missing, so default value of %s will be used", DateHeader.getName(), DateHeader.getValue());
   }
 
-  date_header_include_ = utils::StringUtils::StringToBool(dateHeaderStr, date_header_include_);
+    date_header_include_ = utils::StringUtils::toBool(dateHeaderStr).value_or(true);
```

Comment:
This is a subtle code repetition on the default value. If one would change it, but not this part, it would mean we get an incorrect default. If we lack test coverage as well, it would not even show up as an error.

If you don't want to declare the default as `const`, this might be a slight improvement:
```c++
date_header_include_ = utils::StringUtils::toBool(dateHeaderStr).value_or(DateHeader.getValue());
```

Reply:
Same below.

Reply:
Could you please elaborate a bit more on "If one would change it, but not this part, it would mean we get an incorrect default.".